configure.ac

```diff
@@ -94,6 +91,11 @@ case "$with_arch" in
 	AM_LDFLAGS="-nostdlib"
 	AM_CCASFLAGS="-fno-inline-functions -nostdlib -mlongcalls"
 
+	# GCC needs these additional flags on top of any user flags.
+	CFLAGS="${CFLAGS:+$CFLAGS } -O2 -g -Wall -Werror -Wl,-EL -Wmissing-prototypes"
+	LDFLAGS="${LDFLAGS:+$LDFLAGS } ${AM_LDFLAGS}"
```

Comment:
We shouldn't put AM_* flags in user FLAGS, cos it will cause duplication, flags should be in LDFLAGS if is forced or in AM_LDFLAGS otherwise.

Reply:
AC_PROG_CC needs correct CFLAGS etc otherwise compiler is sane check fails. AC_PROG_CC needs and expects a target C library (which we don't need).

Reply:
CFLAGS are ok I meant LDFLAGS/CCASFLAGS ;)

Reply:
ok, can you fix them and add to your unit test build script PR.